src/main/java/ch/njol/skript/effects/EffAdventureRestrictions.java

```diff
@@ -0,0 +1,115 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+@Name("Apply Adventure Restrictions")
+@Description("Allow or prevent an item to destroy or be placed on certain types of blocks while in /gamemode adventure.")
+@Examples({
+	"allow player's tool to destroy (stone, oak wood planks)",
+	"prevent {_item} from being placed on (diamond ore, diamond block)"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class EffAdventureRestrictions extends Effect {
+
+	static {
+		if (Skript.methodExists(ItemMeta.class, "setDestroyableKeys", Collection.class)) {
+			Skript.registerEffect(EffAdventureRestrictions.class,
+				"allow %~itemtypes% to (destroy|break|mine|place:be placed on) %itemtypes% [in adventure [mode]]",
+				"(disallow|prevent) %~itemtypes% from (destroying|breaking|mining|place:being placed on) %itemtypes% [in adventure [mode]]");
```

Comment:
would like some opinions on whether this should require `in adventure [mode]` or not

Reply:
Well yes(*)
It's going to get confused with suitable tools (pickaxe -> stone, axe -> wood) otherwise, since people will assume that a shovel can break dirt but that's not true in this condition.

Reply:
Yeah I'm definitely in favour of having the `in adventure [mode]` part, just for clarity.

Reply:
Committed new examples which match the current patterns. I'll leave this conversation open in case anyone wants to discuss this further.

Reply:
yeah make it required